src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -1994,6 +1997,112 @@ namespace Partition {
+    const Location &/*loc*/, Vec<ASR::expr_t*>& /*args*/) {
+        // TODO
+        return nullptr;
+    }
+
+    static inline ASR::asr_t* create_SymbolicSymbol(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() != 1) {
+            err("Intrinsic Symbol function accepts exactly 1 argument", loc);
+        }
+
+        ASR::ttype_t *type = ASRUtils::expr_type(args[0]);
+        if (!ASRUtils::is_character(*type)) {
+            err("Argument of the Symbol function must be a Character",
+                args[0]->base.loc);
+        }
+
+        ASR::ttype_t *to_type = ASRUtils::TYPE(ASR::make_SymbolicExpression_t(al, loc));
+        return UnaryIntrinsicFunction::create_UnaryFunction(al, loc, args, eval_SymbolicSymbol,
+            static_cast<int64_t>(ASRUtils::IntrinsicFunctions::SymbolicSymbol), 0, to_type);
+    }
+
+} // namespace SymbolicSymbol
+
+namespace SymbolicAdd {
+
+    static inline void verify_args(const ASR::IntrinsicFunction_t& x, diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args == 2, "SymbolicAdd must have exactly two arguments",
+            x.base.base.loc, diagnostics);
+
+        ASR::ttype_t* left_type = ASRUtils::expr_type(x.m_args[0]);
+        ASR::ttype_t* right_type = ASRUtils::expr_type(x.m_args[1]);
+
+        ASRUtils::require_impl(ASR::is_a<ASR::SymbolicExpression_t>(*left_type) &&
+            ASR::is_a<ASR::SymbolicExpression_t>(*right_type),
+            "Both arguments of SymbolicAdd must be of type SymbolicExpression",
+            x.base.base.loc, diagnostics);
+    }
+
+    static inline ASR::expr_t *eval_SymbolicAdd(Allocator &/*al*/,
+    const Location &/*loc*/, Vec<ASR::expr_t*>& /*args*/) {
+        // TODO
+        return nullptr;
+    }
+
+    static inline ASR::asr_t* create_SymbolicAdd(Allocator& al, const Location& loc,
+            Vec<ASR::expr_t*>& args,
+            const std::function<void (const std::string &, const Location &)> err) {
+        if (args.size() != 2) {
+            err("Intrinsic Symbol Add operator accepts exactly 2 arguments", loc);
+        }
+
+        Vec<ASR::expr_t*> arg_values;
+        arg_values.reserve(al, args.size());
+        for( size_t i = 0; i < args.size(); i++ ) {
+            arg_values.push_back(al, ASRUtils::expr_value(args[i]));
+        }
+        ASR::expr_t* compile_time_value = eval_SymbolicAdd(al, loc, arg_values);
+        ASR::ttype_t *to_type = ASRUtils::TYPE(ASR::make_SymbolicExpression_t(al, loc));
+        return ASR::make_IntrinsicFunction_t(al, loc,
+                static_cast<int64_t>(ASRUtils::IntrinsicFunctions::SymbolicAdd),
+                args.p, args.size(), 0, to_type, compile_time_value);
+    }
+
+} // namespace SymbolicAdd
+
+namespace SymbolicPi {
+
+    static inline void verify_args(const ASR::IntrinsicFunction_t& /*x*/, diag::Diagnostics& /*diagnostics*/) {
```

Comment:
If it does not take arguments, I think we verify here that
```
ASRUtils::require_impl(x.n_args == 0, "SymbolicPi does not take arguments",
            x.base.base.loc, diagnostics);
```

PS: I think it can be done separately

Reply:
Yes, this is a good catch , could be added to `verify_args`

Reply:
Resolving as addressed .